packages/vite/src/node/server/index.ts

```diff
@@ -396,16 +401,32 @@ export async function createServer(
       }
     },
     async listen(port?: number, isRestart?: boolean) {
-      await startServer(server, port, isRestart)
+      await startServer(server, port)
       if (httpServer) {
         server.resolvedUrls = await resolveServerUrls(
           httpServer,
           config.server,
           config,
         )
       }
+
+      if (!isRestart && config.server.open) server.openDevBrowser()
```

Comment:
I think `server.openBrowser` and the if condition, should be moved inside the `if (httpServer)` block so that `openBrowser` doesn't warn that there's no URL.

Reply:
very good